File: src/main/java/org/jukeboxmc/form/element/ImageButton.java
package org.jukeboxmc.form.element;

import org.jetbrains.annotations.NotNull;
import org.json.simple.JSONObject;

/**
 * @author GoMint
 * @version 1.0
 */
public class ImageButton extends Button {

    private final String image;

    public ImageButton( String id, String text, String image ) {
        super( id, text );
        this.image = image;
    }

    @Override
    public @NotNull JSONObject toJSON() {
        JSONObject button = super.toJSON();
        JSONObject jsonIcon = new JSONObject();
        jsonIcon.put( "type", this.image.startsWith( "http" ) || this.image.startsWith( "https" ) ? "url" : "path" );
        jsonIcon.put( "data", this.image );
        button.put( "image", jsonIcon );
        return button;
    }

}

File: src/main/java/org/jukeboxmc/form/element/ImageButton.kt
package org.jukeboxmc.form.element

import org.json.simple.JSONObject

/**
 * @author GoMint
 * @version 1.0
 */
class ImageButton(id: String?, text: String?, private val image: String) : Button(id, text) {
    override fun toJSON(): JSONObject {
        val button = super.toJSON()
        val jsonIcon = JSONObject()
        jsonIcon["type"] = if (image.startsWith("http") || image.startsWith("https")) "url" else "path"
        jsonIcon["data"] = image
        button!!["image"] = jsonIcon
        return button
    }
}